### PR TITLE
Some bugfix

### PR DESCRIPTION
**Pull request by yang1666204**
## Summary

1. Fix the issue that role is not updated.
2. Verify that the role exists when creating it.
3. Support administrators to modify other users' passwords.
4. Deleting a role that is already in use by a user is not allowed.

### CODE REVIEW

File: ui/src/pages/Access/index.tsx [hunk 1]
```diff
@@ -58,6 +66,16 @@ export default function Access() {
     }
   };
 
+  const existingRole = useMemo(() => {
```
Comment (powerfooI) [inline ui/src/pages/Access/index.tsx:69]: Naming the variable to plural form like `existingRoles` would be better.

File: ui/src/pages/Access/Roles.tsx [hunk 2]
```diff
@@ -120,6 +130,16 @@ export default function Roles({ allRoles, refreshRoles }: RolesProps) {
     setModalVisible(true);
   };
 
+  const existingRole = useMemo(() => {
```
Comment (powerfooI) [inline ui/src/pages/Access/Roles.tsx:133]: Why not pass the `existingRole` from `Access/index` to `Access/Roles`? Actually you've calculated one in parent component.

File: ui/src/pages/Access/Accounts.tsx [hunk 3]
```diff
@@ -122,7 +125,18 @@ export default function Accounts({
         return (
           <Space>
             <Button
-              onClick={() => setResetModalVisible(true)}
+              onClick={() => {
```
[the remark this answers is not in the document]
Comment (yang1666204) [reply ui/src/pages/Access/Accounts.tsx:128]: Okey.

File: ui/src/pages/Access/index.tsx (hunk 1 above)
Comment (yang1666204) [reply ui/src/pages/Access/index.tsx]: Done.

File: ui/src/pages/Access/Roles.tsx (hunk 2 above)
Comment (yang1666204) [reply ui/src/pages/Access/Roles.tsx]: Done.

**Review by powerfooI — APPROVED** [review]
LGTM

**Pull request merged**